File: repos.rb
#!/usr/bin/ruby

require 'open-uri'
require 'json'


10.times do |n|

  contents = URI.parse("https://api.github.com/legacy/repos/search/:java?language=java&start_page=#{n}").read

  result = JSON.parse(contents)

  result.each { |item|
    puts item.url
  }

end

File: repos.py
#!/usr/bin/python

import urllib2
import json

for n in range(11):
	contents = urllib2.urlopen("https://api.github.com/legacy/repos/search/:java?language=java&start_page=%s" % n).read()

	j = json.loads(contents)

	for item in j['repositories']:
		print item['url']
